rs/src/output/version.rs

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module for the version number struct.
+
+use crate::output::diagnostic;
+
+/// A version number.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Version(pub u32, pub u32, pub u32);
+
+impl std::fmt::Display for Version {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}.{}.{}", self.0, self.1, self.2)
+    }
+}
+
+impl std::str::FromStr for Version {
```

Comment:
Can't you use something like https://github.com/dtolnay/semver for this?

Reply:
I didn't really think to use something preexisting to be honest. I'll have a look. Substrait version compatibility does not match semver exactly though.

Reply:
Tried something in 1b0da4e, not sure if idiomatic.

Reply:
Very nice.